### PR TITLE
Add newly supported Linux/MacOS SOS commands

### PR DESCRIPTION
**Pull request by mikem8361**
Fixes issue: https://github.com/dotnet/diagnostics/issues/3395

### CODE REVIEW

File: docs/core/diagnostics/dotnet-dump.md [hunk 1]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
-| `histroot <arguments>`                              | Displays information related to both promotions and relocations of the specified root.        |
-| `lm` or `modules`                                   | Displays the native modules in the process.                                                   |
-| `name2ee <arguments>`                               | Displays the `MethodTable` and `EEClass` structures for the `<argument>`.                     |
-| `pe <arguments>` or `printexception <arguments>`    | Displays any object derived from the <xref:System.Exception> class for the `<argument>`.      |
-| `setsymbolserver <arguments>`                       | Enables the symbol server support                                                             |
-| `syncblk <arguments>`                               | Displays the SyncBlock holder info.                                                           |
-| `threads <threadid>` or `setthread <threadid>`      | Sets or displays the current thread ID for the SOS commands.                                  |
+| `soshelp` or `help`                                 | Displays all available commands
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:152]:
```suggestion
| `soshelp` or `help`                                 | Displays all available commands.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 2]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `soshelp <command>` or `help <command>`             | Displays the specified command.
+| `exit` or `quit`                                    | Exits interactive mode.
+| `analyzeoom`                                        | Displays the info of the last OOM occurred on an allocation request to the GC heap.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:155]:
```suggestion
| `analyzeoom`                                        | Displays the info of the last OOM that occurred on an allocation request to the GC heap.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 3]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `clrmodules`                                        | Lists the managed modules in the process.
+| `clrstack`                                          | Provides a stack trace of managed code only.
+| `clrthreads`                                        | Lists the managed threads that are running.
+| `clru`                                              | Displays an annotated disassembly of a managed method.
+| `d` or `readmemory`                                 | Dumps memory contents.
+| `dbgout`                                            | Enables/disables (-off) internal SOS logging.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:161]:
```suggestion
| `dbgout`                                            | Enables/disables (`-off`) internal SOS logging.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 4]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `dso`                                               | Displays all managed objects found within the bounds of the current stack.
+| `dumpalc`                                           | Displays details about a collectible AssemblyLoadContext to which the specified object is loaded.
+| `dumparray`                                         | Displays details about a managed array.
+| `dumpasync`                                         | Displays info about async state machines on the garbage-collected heap.
+| `dumpassembly`                                      | Displays details about an assembly.
+| `dumpclass`                                         | Displays information about the `EEClass` structure at the specified address.
+| `dumpconcurrentdictionary`                          | Displays concurrent dictionary content.
+| `dumpconcurrentqueue`                               | Displays concurrent queue content.
+| `dumpdelegate`                                      | Displays information about a delegate.
+| `dumpdomain`                                        | Displays information all the AppDomains and all assemblies within the specified domain.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:171]:
```suggestion
| `dumpdomain`                                        | Displays information about the assemblies within the specified AppDomain.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 5]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `dumpgcdata`                                        | Displays information about the GC data.
+| `dumpgen`                                           | Displays heap content for the specified generation.
+| `dumpheap`                                          | Displays info about the garbage-collected heap and collection statistics about objects.
+| `dumpil`                                            | Displays the Microsoft intermediate language (MSIL) that is associated with a managed method.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:175]:
```suggestion
| `dumpil`                                            | Displays the Microsoft intermediate language (MSIL) that's associated with a managed method.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 6]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `dumplog`                                           | Writes the contents of an in-memory stress log to the specified file.
+| `dumpmd`                                            | Displays information about the `MethodDesc` structure at the specified address.
+| `dumpmodule`                                        | Displays information about the module at the specified address.
+| `dumpmt`                                            | Displays information about the method table at the specified address.
+| `dumpobj`                                           | Displays info the object at the specified address.
+| `dumpruntimetypes`                                  | Finds all System.RuntimeType objects in the gc heap and prints the type name and MethodTable they refer too.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:181]:
```suggestion
| `dumpruntimetypes`                                  | Finds all System.RuntimeType objects in the GC heap and prints the type name and MethodTable they refer too.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 7]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `dumpsig`                                           | This command dumps the signature of a method or field given by \<sigaddr\> \<moduleaddr\>.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:182]:
```suggestion
| `dumpsig`                                           | Dumps the signature of a method or field specified by `<sigaddr> <moduleaddr>`.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 8]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `dumpsigelem`                                       | This command dumps a single element of a signature object.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:183]:
```suggestion
| `dumpsigelem`                                       | Dumps a single element of a signature object.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 9]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `dumpstack`                                         | Displays a native and managed stack trace.
+| `dumpstackobjects`                                  | Displays all managed objects found within the bounds of the current stack.
+| `dumpvc`                                            | Displays info about the fields of a value class.
+| `eeheap`                                            | Displays info about process memory consumed by internal runtime data structures.
+| `eestack`                                           | Runs `dumpstack` on all threads in the process.
+| `eeversion`                                         | Displays information about the runtime and SOS versions.
+| `ehinfo`                                            | Displays the exception handling blocks in a jitted method.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:190]:
```suggestion
| `ehinfo`                                            | Displays the exception handling blocks in a JIT-ted method.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 10]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `finalizequeue`                                     | Displays all objects registered for finalization.
+| `findappdomain`                                     | Attempts to resolve the AppDomain of a GC object.
+| `gchandles`                                         | Displays statistics about garbage collector handles in the process.
+| `gcheapstat`                                        | Displays statistics about garbage collector.
+| `gcinfo`                                            | Displays info JIT GC encoding for a method.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:195]:
```suggestion
| `gcinfo`                                            | Displays the JIT GC encoding for a method.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 11]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `gcroot`                                            | Displays info about references (or roots) to the object at the specified address.
+| `gcwhere`                                           | Displays the location in the GC heap of the argument passed in.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:197]:
```suggestion
| `gcwhere`                                           | Displays the location in the GC heap of the specified address.
```

Comment (gewarren) [reply docs/core/diagnostics/dotnet-dump.md:197]: Not sure if address is correct though.

File: docs/core/diagnostics/dotnet-dump.md [hunk 12]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `histclear`                                         | Releases any resources used by the family of Hist commands.
+| `histinit`                                          | Initializes the SOS structures from the stress log saved in the debuggee.
+| `histobj`                                           | Examines all stress log relocation records and displays the chain of garbage collection relocations that may have led to the address passed in as an argument.
+| `histobjfind`                                       | Displays all the log entries that reference the object at the specified address.
+| `histroot`                                          | Displays information related to both promotions and relocations of the specified root.
+| `histstats`                                         | Displays stress log stats.
+| `ip2md`                                             | Displays the `MethodDesc` structure at the specified address in code that has been JIT-compiled.
+| `listnearobj`                                       | Displays the object preceeding and succeeding the address passed.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:205]:
```suggestion
| `listnearobj`                                       | Displays the object preceding and succeeding the specified address.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 13]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `logopen`                                           | Enables console file logging.
+| `logclose`                                          | Disables console file logging.
+| `logging`                                           | Enables/disables internal SOS logging.
+| `lm` or `modules`                                   | Displays the native modules in the process.
+| `name2ee`                                           | Displays the `MethodTable` and `EEClass` structures for the specified type or method in the specified module.
+| `objsize`                                           | Displays the size of the specified object.
+| `parallelstacks`                                    | Displays merged threads stack a la Visual Studio 'Parallel Stacks' panel.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:212]:
```suggestion
| `parallelstacks`                                    | Displays the merged threads stack similarly to the Visual Studio 'Parallel Stacks' panel.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 14]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `pathto`                                            | Displays the GC path from \<root\> to \<target\>.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:213]:
```suggestion
| `pathto`                                            | Displays the GC path from `<root>` to `<target>`.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 15]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `pe` or `printexception`                            | Displays and formats fields of any object derived from the <xref:System.Exception> class at the specified address.
+| `r` or `registers`                                  | Displays the thread's registers.
+| `runtimes`                                          | Lists the runtimes in the target or change the default runtime.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:216]:
```suggestion
| `runtimes`                                          | Lists the runtimes in the target or changes the default runtime.
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 16]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `setclrpath`                                        | Sets the path to load coreclr dac/dbi files. `setclrpath <path>`
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:217]:
```suggestion
| `setclrpath`                                        | Sets the path to load coreclr dac/dbi files using `setclrpath <path>`
```

File: docs/core/diagnostics/dotnet-dump.md [hunk 17]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `setsymbolserver`                                   | Enables the symbol server support.
+| `sos`                                               | Various coreclr debugging commands. For more information, see 'soshelp'. `sos <command-name> <args>`
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:219]:
```suggestion
| `sos`                                               | Executes various coreclr debugging commands. Use the syntax `sos <command-name> <args>`. For more information, see 'soshelp'.
```

Comment (gewarren) [reply docs/core/diagnostics/dotnet-dump.md:219]: Where is 'soshelp'?

File: docs/core/diagnostics/dotnet-dump.md [hunk 18]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
+| `syncblk`                                           | Displays the SyncBlock holder info.
+| `taskstate`                                         | Displays a Task state in a human readable format.
+| `threadpool`                                        | Displays info about the runtime thread pool.
+| `threadpoolqueue`                                   | Displays queued thread pool work items.
+| `threadstate`                                       | Pretty prints the meaning of a threads state.
+| `threads <threadid>` or `setthread <threadid>`      | Sets or displays the current thread ID for the SOS commands.
+| `timerinfo`                                         | Displays running timers details.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:226]:
```suggestion
| `timerinfo`                                         | Displays information about running timers.
```

File: docs/core/diagnostics/sos-debugging-extension.md [hunk 19]
```diff
@@ -121,41 +121,84 @@ By default you can reach all the SOS commands by entering: `sos [command\_name]`
 
 | Command                               | Function
 | ------------------------------------- | ---------------------------------------------------------------------------------------------
+|    `analyzeoom`                       | Displays the info of the last OOM occurred on an allocation request to the GC heap.
```
Comment (gewarren) [inline docs/core/diagnostics/sos-debugging-extension.md:124]:
```suggestion
|    `analyzeoom`                       | Displays the info of the last OOM that occurred on an allocation request to the GC heap.
```

**Review by gewarren — COMMENTED** [review]
Would be good to apply the suggestions to the second file too but I didn't repeat the suggestions there.

**Comment by mikem8361** [comment]
Yes, I'll apply these changes to the second doc.

**Review by noahfalk — APPROVED** [review]
LGTM 👍

File: docs/core/diagnostics/dotnet-dump.md [hunk 20]
```diff
@@ -149,40 +149,85 @@ dotnet-dump analyze <dump_path> [-h|--help] [-c|--command]
...
-| `histroot <arguments>`                              | Displays information related to both promotions and relocations of the specified root.        |
-| `lm` or `modules`                                   | Displays the native modules in the process.                                                   |
-| `name2ee <arguments>`                               | Displays the `MethodTable` and `EEClass` structures for the `<argument>`.                     |
-| `pe <arguments>` or `printexception <arguments>`    | Displays any object derived from the <xref:System.Exception> class for the `<argument>`.      |
-| `setsymbolserver <arguments>`                       | Enables the symbol server support                                                             |
-| `syncblk <arguments>`                               | Displays the SyncBlock holder info.                                                           |
-| `threads <threadid>` or `setthread <threadid>`      | Sets or displays the current thread ID for the SOS commands.                                  |
+| `soshelp` or `help`                                 | Displays all available commands.
```
Comment (gewarren) [inline docs/core/diagnostics/dotnet-dump.md:152]: Shouldn't they all be alphabetical?

**Review by gewarren — APPROVED** [review]

**Pull request merged**